### ISSUE TITLE
Add support for 'params' method arguments

### ISSUE THREAD

**Issue by VitaliyMF** [nreco/lambdaparser#3]
It is useful to support method calls with variable number of parameters (in c# marked as 'params').
Good example is String.Format.

**Comment by EricB85** [comment]
It would be very useful indeed (to me anyway). Did you find how to do it?

**Comment by VitaliyMF** [comment]
Real method argument is an array, so you can pass variable number of arguments by composing an array:

`Format("test {0} {1}", new [] { "argument0", "argument1"} )`

**Comment by EricB85** [comment]
But how would you do a method like "DoSometing(t1,t2)" and "DoSometing(t1, t2, t3, t4)" ?

**Comment by VitaliyMF** [comment]
It looks like it is possible to handle "params" argument in the same way as it works in C# (when compiler automatically converts method arguments into array): https://stackoverflow.com/questions/627656/determining-if-a-parameter-uses-params-using-reflection-in-c

To handle "params" method argument some modifications are needed here: https://github.com/nreco/lambdaparser/blob/master/src/NReco.LambdaParser/InvokeMethod.cs

**Comment by microspud** [comment]
Hi, I have coded the param feature tonight and it appears to work fine. I have tested several different scenarios with different types, mixed types etc and all appears to work fine including Format (via a proxy call in the test object). I will create a separate pull request for it for review.

**Comment by hzy-6** [comment]
Params does not support

SDK：Net7.0

![image](https://github.com/nreco/lambdaparser/assets/14025507/b69e0cff-5d63-4d35-bbc5-70eb17324262)
![image](https://github.com/nreco/lambdaparser/assets/14025507/926f52d0-66c0-43dc-a749-6581ad8bbe8f)

**Comment by VitaliyMF** [comment]
@hzy-6 to enable support of 'params' and optional parameters you need to replace default `InvokeMethod` with `OptionsParamsInvokeMethod`:

```
var lambdaParser = new LambdaParser(OptionsParamsInvokeMethod.Instance);
```

**Comment by hzy-6** [comment]
May I ask how to implement this sum function

![image](https://github.com/nreco/lambdaparser/assets/14025507/75adaf56-ae8b-4048-b0be-b035d2edddec)

**Comment by microspud** [comment]
@hzy-6 I cant see what is in your GetContext() method but you need to add an instance of your Funcs class with a name such as "f" to the context object.

Then in your expression use "f.sum(1,2,3...)

**Comment by VitaliyMF** [comment]
@hzy-6 

```
	public class Funcs {
		public object Sum(params object[] args) {
			decimal res = 0;
			foreach (var a in args)
				res += Convert.ToDecimal(a);
			return res;
		}
	}

	var lambdaParser = new LambdaParser(OptionsParamsInvokeMethod.Instance);
	var varContext = new Dictionary<string, object>();
	varContext["f"] = new Funcs();
	Assert.Equal(6M, lambdaParser.Eval("f.Sum(1,2,3)", varContext));
```

**Comment by hzy-6** [comment]
I hope there is no "f."

**Comment by hzy-6** [comment]
Hope to adapt to existing formula libraries
![image](https://github.com/nreco/lambdaparser/assets/14025507/41298b37-e038-4d44-9817-d57b7b991920)

**Comment by VitaliyMF** [comment]
> I hope there is no "f."

No, currently "params" are supported only for object's method invocation, not for delegates.
"sum" can be a delegate, but delegates are invoked differently in `LambdaParameterWrapper.InvokeDelegate` method.

@microspud I didn't tested this yet, but technically delegate declarations also may have params for instance:
```
public delegate object SumFunc(params object[] args);
```
so, potentially, we can also add support for "params" in delegates by replacing ` deleg.DynamicInvoke(resolvedArgs)` with `IInvokeMethod.InvokeDelegate` (that can be added into our IInvokeMethod interface). What do you think?

**Comment by hzy-6** [comment]
Thank you. What exactly should we do

**Comment by hzy-6** [comment]
Unable to intercept

![image](https://github.com/nreco/lambdaparser/assets/14025507/2f52fa6f-e8f2-4f9b-a8f6-a03f14fe1617)

**Comment by VitaliyMF** [comment]
> Unable to intercept

To support "params" for delegates invocations NReco.LambdaParser code needs to be changed, you cannot achieve that in the current version. 

If you know what to do, you can make necessary changes and propose a PR. Otherwise, you need to wait until @microspud or me finds a time for this change.

**Comment by hzy-6** [comment]
Okay, thank you. I'll learn it first. I hope we can release the function interceptor later
